patches/0013-start-arm-optimized-nistz256

```diff
@@ -0,0 +1,2577 @@
+Index: aws-lc/third_party/boringssl/crypto/fipsmodule/CMakeLists.txt
+===================================================================
+--- aws-lc.orig/third_party/boringssl/crypto/fipsmodule/CMakeLists.txt
++++ aws-lc/third_party/boringssl/crypto/fipsmodule/CMakeLists.txt
+@@ -64,6 +64,7 @@ if(${ARCH} STREQUAL "aarch64")
+     armv8-mont.${ASM_EXT}
+     ghash-neon-armv8.${ASM_EXT}
+     ghashv8-armx.${ASM_EXT}
++    p256-armv8-asm.${ASM_EXT}
+     sha1-armv8.${ASM_EXT}
+     sha256-armv8.${ASM_EXT}
+     sha512-armv8.${ASM_EXT}
+@@ -102,6 +103,7 @@ perlasm(md5-586.${ASM_EXT} md5/asm/md5-5
+ perlasm(md5-x86_64.${ASM_EXT} md5/asm/md5-x86_64.pl)
+ perlasm(p256-x86_64-asm.${ASM_EXT} ec/asm/p256-x86_64-asm.pl)
+ perlasm(p256_beeu-x86_64-asm.${ASM_EXT} ec/asm/p256_beeu-x86_64-asm.pl)
++perlasm(p256-armv8-asm.${ASM_EXT} ec/asm/p256-armv8-asm.pl)
+ perlasm(rdrand-x86_64.${ASM_EXT} rand/asm/rdrand-x86_64.pl)
+ perlasm(rsaz-avx2.${ASM_EXT} bn/asm/rsaz-avx2.pl)
+ perlasm(sha1-586.${ASM_EXT} sha/asm/sha1-586.pl)
+Index: aws-lc/third_party/boringssl/crypto/fipsmodule/ec/ec.c
+===================================================================
+--- aws-lc.orig/third_party/boringssl/crypto/fipsmodule/ec/ec.c
++++ aws-lc/third_party/boringssl/crypto/fipsmodule/ec/ec.c
+@@ -249,6 +249,9 @@ DEFINE_METHOD_FUNCTION(struct built_in_c
+ #if !defined(OPENSSL_NO_ASM) && defined(OPENSSL_X86_64) && \
+     !defined(OPENSSL_SMALL)
+       EC_GFp_nistz256_method();
++#elif !defined(OPENSSL_NO_ASM) && defined(OPENSSL_AARCH64) && \
++    !defined(OPENSSL_SMALL)
++      EC_GFp_nistz256_arm_method();
```

Comment:
Should we name the x86_64 EC_METHOD for `EC_GFp_nistz256_x86_64_method`?

Reply:
We should end up with this method being used both with x86_64 and ARMv8